app/src/utils/DataFetcher.ts

```diff
@@ -46,6 +48,65 @@ class DataFetcher {
     const data = await this.loadData(url);
     return this.convertJson(data);
   }
+
+  async getDatasets() {
```

Comment:
Couple things:
- can we rename this to getDatasetMetadata, or getMetadata?
- my preference is for this to return a map of id to metadata (that's how it was specified in the API definition, though definitely up for debate). The reasoning is that I think looking up metadata by id will be a common pattern, and it's easier to convert a map to an array than vice versa
- can you add a return type annotation for clarity? If using a map it would be `Promise<Record<string, DatasetMetadata>>`